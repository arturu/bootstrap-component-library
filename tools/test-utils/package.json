{
  "name": "@openeuropa/bcl-test-utils",
  "version": "0.1.3",
  "description": "A local test helper",
  "main": "index.js",
  "private": true,
  "devDependencies": {
    "deepmerge": "4.2.2"
  }
}
