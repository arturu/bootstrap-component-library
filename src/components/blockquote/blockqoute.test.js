import { merge, renderTwigFileAsNode } from "@oe/test-utils";
import demoData from "@oe/data-blockquote/data";

const template = "@oe/blockquote/blockquote.html.twig";
const render = (params) => renderTwigFileAsNode(template, params);

describe("OE - Blockquote", () => {
  test(`renders correctly`, () => {
    expect.assertions(1);

    return expect(
      render({ ...demoData})
    ).resolves.toMatchSnapshot();
  });

  test(`renders correctly with no cite`, () => {
    expect.assertions(1);

    return expect(
      render({ ...demoData, cite: ''})
    ).resolves.toMatchSnapshot();
  });
  
  test(`renders correctly with no attribution`, () => {
    expect.assertions(1);

    return expect(
      render({ ...demoData, attribution: ''})
    ).resolves.toMatchSnapshot();
  });

  test(`renders correctly with text aligned center`, () => {
    expect.assertions(1);

    return expect(
      render({ ...demoData, alignment: 'center'})
    ).resolves.toMatchSnapshot();
  });
});
