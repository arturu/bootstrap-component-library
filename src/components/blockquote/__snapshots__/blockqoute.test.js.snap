// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OE - Blockquote renders correctly 1`] = `
<figure
  class="text-left"
>
  <blockquote
    class="blockquote"
  >
    <p>
      A well-known quote, contained in a blockquote element.
    </p>
  </blockquote>
  <figcaption
    class="blockquote-footer"
  >
    Someone famous in    
    <cite
      title="Source title"
    >
      Source title
    </cite>
  </figcaption>
</figure>
`;

exports[`OE - Blockquote renders correctly with no attribution 1`] = `
<figure
  class="text-left"
>
  <blockquote
    class="blockquote"
  >
    <p>
      A well-known quote, contained in a blockquote element.
    </p>
  </blockquote>
</figure>
`;

exports[`OE - Blockquote renders correctly with no cite 1`] = `
<figure
  class="text-left"
>
  <blockquote
    class="blockquote"
  >
    <p>
      A well-known quote, contained in a blockquote element.
    </p>
  </blockquote>
  <figcaption
    class="blockquote-footer"
  >
    Someone famous in  
  </figcaption>
</figure>
`;

exports[`OE - Blockquote renders correctly with text aligned center 1`] = `
<figure
  class="text-center"
>
  <blockquote
    class="blockquote"
  >
    <p>
      A well-known quote, contained in a blockquote element.
    </p>
  </blockquote>
  <figcaption
    class="blockquote-footer"
  >
    Someone famous in    
    <cite
      title="Source title"
    >
      Source title
    </cite>
  </figcaption>
</figure>
`;
